metadata-service/auth-filter/src/main/java/com/datahub/auth/authentication/filter/AuthenticationFilter.java

```diff
@@ -103,38 +105,43 @@ protected void doFilterInternal(
       HttpServletRequest request, HttpServletResponse response, FilterChain chain)
       throws ServletException, IOException {
     AuthenticationRequest context = buildAuthContext(request);
-    Authentication authentication = null;
     try {
-      authentication = this.authenticatorChain.authenticate(context, _logAuthenticatorExceptions);
-    } catch (AuthenticationException e) {
-      // For AuthenticationExpiredExceptions, terminate and provide that feedback to the user
-      log.debug(
-          "Failed to authenticate request. Received an AuthenticationExpiredException from authenticator chain.",
-          e);
-      ((HttpServletResponse) response)
-          .sendError(HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized to perform this action.");
-      return;
-    }
-
-    if (authentication != null) {
-      String actorUrnStr = authentication.getActor().toUrnStr();
+      Authentication authentication =
+          Objects.requireNonNull(
+              this.authenticatorChain.authenticate(context, _logAuthenticatorExceptions));
       // Successfully authenticated.
       log.debug(
           "Successfully authenticated request for Actor with type: {}, id: {}",
           authentication.getActor().getType(),
           authentication.getActor().getId());
       AuthenticationContext.setAuthentication(authentication);
       chain.doFilter(request, response);
-    } else {
+
+    } catch (AuthenticationExpiredException e) {
+      // For AuthenticationExpiredExceptions, terminate and provide that feedback to the user
+      log.debug(
+          "Failed to authenticate request. Unauthorized to perform this action due to expired auth.",
+          e);
+      response.sendError(
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "Unauthorized to perform this action due to expired auth.");
+      return;
+    } catch (AuthenticationException e) {
+      log.debug(
+          "Failed to authenticate request. Received an AuthenticationException from authenticator chain.",
+          e);
+      response.sendError(
+          HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized to perform this action.");
+      return;
+    } catch (Exception e) {
       // Reject request
       log.debug(
-          "Failed to authenticate request. Received 'null' Authentication value from authenticator chain.");
-      ((HttpServletResponse) response)
-          .sendError(
-              HttpServletResponse.SC_UNAUTHORIZED,
-              "Unauthorized to perform this action due to expired auth.");
+          "Failed to authenticate request. Received an exception from the authenticator chain.", e);
+      response.sendError(
+          HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized to perform this action.");
```

Comment:
Why not send the exception context in the HTTP response? Present the actual error to the user.

Reply:
Security best practice is to not leak this information to the caller.

Reply:
Sending the exact exception is in general frowned upon as it gives malicious actors information.